src/schemas/subject.rs

```diff
@@ -0,0 +1,42 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+#[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct Subject {
+    /// "The subject's name that is displayed when processing this certificate"
+    pub display_name: String,
+    /// "Contact information of the subject"
+    pub contact: Contact,
+    /// "The legal entity of the subject"
+    pub legal_entity: Option<LegalEntity>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct Contact {
+    /// "Contact email"
+    pub email: String,
+    #[serde(flatten)]
+    pub other: HashMap<String, Value>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct LegalEntity {
+    /// "Name of the legal entity"
+    pub name: Option<String>,
+    /// "Official address of the legal entity"
+    pub address: Option<Address>,
+    #[serde(flatten)]
+    pub other: HashMap<String, Value>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub enum Address {
+    Basic(String),
+    Complex(serde_json::value::Map<String, Value>),
```

Comment:
Schema describes is as either `String` or an `Object`.

Reply:
As per above, we do not need `Address` as we are not definning `LegalEntity` as a type.